src/cargo/ops/cargo_install.rs

```diff
@@ -317,6 +317,14 @@ impl<'gctx> InstallablePackage<'gctx> {
         self.gctx.shell().status("Installing", &self.pkg)?;
 
         let dst = self.root.join("bin").into_path_unlocked();
+        // `dst` is usually absolute; if not, make it absolute so messages are clearer.
+        // See: https://github.com/rust-lang/cargo/issues/16023
```

Comment:
The two duplicate make-absolute code actually fix two different warnings, should we make one commit for each to make the intent more obvious?